File: SeLeAs/LeetCode/Top 100 Liked/invertBinaryTree.java


public class invertBinaryTree {
    class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null; // Base case: empty tree
        }
        
        // Recursively invert the left and right subtrees
        TreeNode left = invertTree(root.left);
        TreeNode right = invertTree(root.right);
        
        // Swap left and right children
        root.left = right;
        root.right = left;
        
        return root;
    }
}

}

File: SeLeAs/LeetCode/Top 100 Liked/invertBinaryTree.py
from typing import Optional

from minimumDepthBinaryTree import TreeNode

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
      
        if root is None:
            return
        root.right, root.left = self.invertTree(root.left), self.invertTree(root.right)
        return root 